packages/family-tree/src/NodeAnalyzer/ClassChildAnalyzer.php

```diff
@@ -24,18 +24,19 @@ public function hasChildClassMethod(ClassReflection $classReflection, string $me
     {
         $childrenClassReflections = $this->familyRelationsAnalyzer->getChildrenOfClassReflection($classReflection);
 
-        foreach ($childrenClassReflections as $childClassReflection) {
-            if (! $childClassReflection->hasMethod($methodName)) {
+        foreach ($childrenClassReflections as $childrenClassReflection) {
+            $childrenClassReflectionHasMethod = $childrenClassReflection->hasMethod($methodName);
```

Comment:
This method should be inlined, and not on own line. It returns bool

Reply:
this seems automatically moved by rector as not from This type

Reply:
The bool return should work for both `ThisType` and `ObjectType`.

```php
if ($this->is...()) {
}


if ($item->is...()) {
}
```

I've noticed similar behavior on any variable rename. It seems the variable looses type and returns `MixedType`. That's why it's changed by another Rector rule like this.

It should not be changed.

Reply:
I will check.

Reply:
updated.